ethcore/src/state/mod.rs

```diff
@@ -108,6 +108,20 @@ impl AccountEntry {
 			state: AccountState::Clean,
 		}
 	}
+
+	// Replace data with another entry but preserve storage cache
+	fn merge_snapshot(&mut self, other: AccountEntry) {
+		self.state = other.state;
```

Comment:
Its the other way round. `other` contains "new" changes that replace `self`. Everything in `self` is discarded except for the storage cache